### PR TITLE
feat(api,hooks): add scheduler on workflow v2

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/hooks/scheduler_v2.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+func (s *Service) schedulerExecutionRoutine(ctx context.Context) {
+	tick := time.NewTicker(time.Duration(10) * time.Second)
+	defer tick.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			log.Error(ctx, "schedulerExecutionRoutine > exiting goroutine: %v", ctx.Err())
+			return
+		case <-tick.C:
+			schedulerExecutions, err := s.Dao.GetAllSchedulerExecutions(ctx)
```
Comment (yesnault) [inline engine/hooks/scheduler_v2.go:119]: need maintenance aware

**Review by yesnault — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#20726.0 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#20726.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#20726.1 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

File: engine/hooks/scheduler_v2.go (hunk 1 above)
Comment (sguiheux) [reply engine/hooks/scheduler_v2.go:125]: fixed

**Review by yesnault — APPROVED** [review]

File: engine/api/repositoriesmanager/repositories_manager.go [hunk 2]
```diff
@@ -259,6 +259,9 @@ func (c *vcsClient) Branches(ctx context.Context, fullname string, filters sdk.V
 
 	branches := []sdk.VCSBranch{}
 	path := fmt.Sprintf("/vcs/%s/repos/%s/branches?limit=%d", c.name, fullname, filters.Limit)
+	if filters.NoCache {
+		path += fmt.Sprintf("&noCache=%v", filters.NoCache)
```
Comment (richardlt) [inline engine/api/repositoriesmanager/repositories_manager.go:263]:
```suggestion
		path += fmt.Sprintf("&noCache=%t", filters.NoCache)
```

File: engine/api/repositoriesmanager/repositories_manager.go [hunk 3]
```diff
@@ -280,6 +283,9 @@ func (c *vcsClient) SearchPullRequest(ctx context.Context, repoFullName, commit,
 func (c *vcsClient) Branch(ctx context.Context, fullname string, filters sdk.VCSBranchFilters) (*sdk.VCSBranch, error) {
 	branch := sdk.VCSBranch{}
 	path := fmt.Sprintf("/vcs/%s/repos/%s/branches/?branch=%s&default=%v", c.name, fullname, url.QueryEscape(filters.BranchName), filters.Default)
+	if filters.NoCache {
+		path += fmt.Sprintf("&noCache=%v", filters.NoCache)
```
Comment (richardlt) [inline engine/api/repositoriesmanager/repositories_manager.go:287]:
```suggestion
		path += fmt.Sprintf("&noCache=%t", filters.NoCache)
```

File: engine/api/v2_project_clean_ascode.go [hunk 4]
```diff
@@ -87,6 +93,14 @@ func workerCleanProject(ctx context.Context, db *gorp.DbMap, store cache.Store,
 }
 
 func cleanAscodeProject(ctx context.Context, db *gorp.DbMap, store cache.Store, pKey string, entityRetention time.Duration) error {
+	hookServices, err := services.LoadAllByType(ctx, db, sdk.TypeHooks)
+	if err != nil {
+		return err
+	}
+	if len(hookServices) < 1 {
+		return sdk.NewErrorFrom(sdk.ErrUnknownError, "unable to find 1 hook service")
```
Comment (richardlt) [inline engine/api/v2_project_clean_ascode.go:101]:
```suggestion
		return sdk.NewErrorFrom(sdk.ErrUnknownError, "unable to find hook service")
```

File: engine/api/v2_repository_analyze.go [hunk 5]
```diff
@@ -781,20 +802,37 @@ skipEntity:
 		event_v2.PublishEntityEvent(ctx, api.Cache, sdk.EventEntityDeleted, vcsProjectWithSecret.Name, repo.Name, eRemoved, &userDB)
 	}
 
+	if len(schedulers) != 0 {
+		// Instantiate Schedulers on hooks µservice
+		if _, _, err := services.NewClient(srvs).DoJSONRequest(ctx, http.MethodPost, "/v2/workflow/scheduler", schedulers, nil); err != nil {
+			return api.stopAnalysis(ctx, analysis, sdk.NewErrorFrom(sdk.ErrUnknownError, "unable to instantiate scheduler on hook service"))
```
Comment (richardlt) [inline engine/api/v2_repository_analyze.go:808]: Log src error

**Review by richardlt — CHANGES_REQUESTED** [review]

File: engine/api/v2_repository_analyze.go (hunk 5 above)
Comment (sguiheux) [reply engine/api/v2_repository_analyze.go:812]: fixed

**Comment by ovh-cds** [comment]
CDS Report build-cds#20761.0 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#20761.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by fsamin — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Pull request merged**